src/components/templates/Search/filterPrice.tsx

```diff
@@ -45,52 +49,156 @@ export default function FilterPrice({
     navigate(urlLocation)
   }
 
-  async function applyTypeFilter(filterBy: string) {
+  async function applyServiceFilter(filterBy: string) {
     let urlLocation = await addExistingParamsToUrl(location, 'serviceType')
-    if (filterBy) {
+    if (filterBy && location.search.indexOf('&serviceType') === -1) {
       urlLocation = `${urlLocation}&serviceType=${filterBy}`
     }
     setServiceType(filterBy)
     navigate(urlLocation)
   }
 
+  async function handleSelectedFilter(isSelected: boolean, value: string) {
+    if (
+      value === FilterByPriceOptions.Fixed ||
+      value === FilterByPriceOptions.Dynamic
+    ) {
+      if (isSelected) {
+        if (clearSelected) {
+          setPriceSelections([])
+        } else {
+          if (priceSelections.length > 1) {
+            // both selected -> select the other one
+            const otherValue = priceFilterItems.find((p) => p.value !== value)
+              .value
+            await applyPriceFilter(otherValue)
+            setPriceSelections([otherValue])
+          } else {
+            // only the current one selected -> deselect it
+            await applyPriceFilter(undefined)
+            setPriceSelections([])
+          }
+        }
+      } else {
+        if (priceSelections.length > 0) {
+          // one already selected -> both selected
+          await applyPriceFilter(FilterByPriceOptions.All)
+          setPriceSelections(priceFilterItems.map((p) => p.value))
+          setClearSelected(false)
+        } else {
+          // none selected -> select
+          await applyPriceFilter(value)
+          setPriceSelections([value])
+          setClearSelected(false)
+        }
+      }
+    } else {
+      if (isSelected) {
+        if (clearSelected) {
+          setServiceSelections([])
+        } else {
+          if (serviceSelections.length > 1) {
+            const otherValue = serviceFilterItems.find((p) => p.value !== value)
+              .value
+            await applyServiceFilter(otherValue)
+            setServiceSelections([otherValue])
+          } else {
+            await applyServiceFilter(undefined)
+            setServiceSelections([])
+          }
+        }
+      } else {
+        if (serviceSelections.length) {
+          await applyServiceFilter(undefined)
+          setServiceSelections(serviceFilterItems.map((p) => p.value))
+          setClearSelected(false)
+        } else {
+          await applyServiceFilter(value)
+          setServiceSelections([value])
+          setClearSelected(false)
+        }
+      }
+    }
+  }
+
+  async function applyClearFilter() {
+    let urlLocation = await addExistingParamsToUrl(
+      location,
+      'priceType',
+      'serviceType'
+    )
+
+    urlLocation = `${urlLocation}`
+
+    setServiceSelections([])
+    setPriceSelections([])
+
+    setClearSelected(true)
+    setPriceType(undefined)
+    setServiceType(undefined)
+    navigate(urlLocation)
+  }
+
   return (
     <div>
```

Comment:
for whatever reason, this double `div` was added back in 8cce1898fdd6e0a163c441825d3e8401cf3232c8. Does not look like this is needed